src/arrowc/error.c

```diff
@@ -15,21 +15,28 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#include <errno.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <string.h>
 
 #include "arrowc.h"
 
 int ArrowErrorSet(struct ArrowError* error, const char* fmt, ...) {
-  memset(error->message, 0, 1024);
+  memset(error->message, 0, sizeof(error->message));
 
   va_list args;
   va_start(args, fmt);
-  int result = vsnprintf(error->message, 1024 - 1, fmt, args);
+  int chars_needed = vsnprintf(error->message, sizeof(error->message) - 1, fmt, args);
```

Comment:
Why the -1? Seems vsnprintf will already restrict itself to `bufsz - 1` characters to leave room for a null terminator